Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp

```diff
@@ -54,7 +54,7 @@ template<class T>
 inline static std::optional<RenderingResourceIdentifier> applyImageBufferItem(GraphicsContext& context, const ResourceHeap& resourceHeap, const T& item)
 {
     auto resourceIdentifier = item.imageBufferIdentifier();
-    if (auto* imageBuffer = resourceHeap.getImageBuffer(resourceIdentifier)) {
+    if (RefPtr imageBuffer = resourceHeap.getImageBuffer(resourceIdentifier)) {
```

Comment:
I do not exactly understand the purpose of this change. Does the clang static analyzer tell you that this RefCounted object might be derefed in the call `item.apply()` below? So you are adding RefPtr to protect the object from being accidentally deleted? But even if it is derefed, which I am sure it does not happen, this object is not referenced after the call to `item.apply()`.

Reply:
The main issue is that some of these do a `dynamicDowncast`, which clang annoyingly doesn't seem to understand should have the same lifetime as its source (as long as it's not modified), so I stuffed the downcast pointer into a `RefPtr` to explicitly add a reference after the function ends.
I may have done this one and other non-downcasting ones for consistency, I could revert these. Or do you object to all `RefPtr` returns? Other suggestions? (And I'll try to think of alternatives.)